packages/financial-templates-lib/src/price-feed/BigQueryPriceFeed.js

```diff
@@ -0,0 +1,219 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const { BigQuery } = require("@google-cloud/bigquery");
+const moment = require("moment");
+const highland = require("highland");
+const { parseFixed } = require("@ethersproject/bignumber");
+
+const client = new BigQuery();
+
+// An implementation of PriceFeedInterface that uses BigQuery to retrieve prices.
+class BigQueryPriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs the BigQueryPriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider from truffle instance to connect to Ethereum network.
+   * @param {Integer} lookback How far in the past the historical prices will be available using getHistoricalPrice.
+   * @param {Function} getTime Returns the current time.
+   * @param {Integer} minTimeBetweenUpdates Min number of seconds between updates. If update() is called again before
+   *      this number of seconds has passed, it will be a no-op.
+   * @param {Number} decimals Number of decimals to use to convert price to wei.
+   */
+  constructor(logger, web3, lookback, getTime, minTimeBetweenUpdates, decimals = 0) {
+    super();
+    this.logger = logger;
+    this.web3 = web3;
+    this.lookback = lookback;
+    this.getTime = getTime;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+    this.toBN = this.web3.utils.toBN;
+
+    this.convertDecimals = number => {
+      // Converts price result to wei and returns price conversion to correct decimals as a big number.
+      return this.toBN(parseFixed(number.toString(), decimals).toString());
+    };
+  }
+
+  getCurrentPrice() {
+    return this.currentPrice;
+  }
+
+  async getHistoricalPrice(time) {
+    if (this.lastUpdateTime === undefined) {
+      return undefined;
+    }
+
+    // Using moment package to changeto UTC and create acceptable format for BQ.
+    const formattedCurrentTime = moment(time)
+      .utc()
+      .format("YYYY-MM-DD HH:mm:ss");
+
+    // 2592000000 is a month time interval
```

Comment:
Do you mean 2592000000 = 30 * 24 * 60 * 60 seconds?

Reply:
Yes, definitely confusing. Changed to use the `moment` calculation so I updated the comment.